### DISCUSSION TITLE
Form Schema stored on API with select fields each with it's on options API URL

### DISCUSSION THREAD

**Discussion by gabriel-kohen-by** [data-driven-forms/react-forms#928] [Need help]
Hi All,

First of all I want to share my excitement as I just ran into this library to drive metadata driven forms.
We'd like to be able to store all form definitions(metadata including all the schema supported by this library) on backend API services. So this schema will be loaded at runtime rather than having it as code in build time.
In addition we want the schema to be able to support select(dropdown) fields that can get each their possible options from a definition of a URL for that field metadata.
The approach we thought of taking was thinking to serialize to:

1. Load the specific form JSON schema from the API. 
2. Because select fields `loadOption` only support functions and functions are invalid JSON value, AFAIK, we can transform a specific function name which we will implement into an actual function in the used schema. That function could read an extra prop we can have with that field. For instance `loadOptionsUrl`. Using the value of loadOptionsUrl for that field it will load the possible options from individual APIs.

I'd love to hear if you think we're taking the right approach or there is a better way to achieve this. 
Thanks.

**Comment by Hyperkid123** [comment]
Hi. You can use an action mapper to adjust the schema at runtime. https://data-driven-forms.org/mappers/action-mapper#heading-actionmapper

Or create a wrapper around the select component.

I can create some examples tomorrow.

**Reply by rvsia** [reply]
Yeah, I agree with this. The Action Mapper feature was specifically created for cases like this.

**Reply by Hyperkid123** [reply]
@gabriel-kohen-by here is an example using the action mapper: https://codesandbox.io/s/data-driven-forms-example-forked-cp5pd?file=/src/App.js

Apologies in advance if I won't be replying regularly. You know Christmas and all that. :evergreen_tree:
